mmv1/third_party/terraform/fwtransport/framework_config_test.go.erb

```diff
@@ -0,0 +1,393 @@
+			ExpectedConfigStructValue: types.StringValue("project-from-GOOGLE_CLOUD_PROJECT"),
+		},
+		"project value can be set by environment variable: GCLOUD_PROJECT is used third": {
+			ConfigValues: fwmodels.ProviderModel{
+				Project: types.StringNull(), // unset
+			},
+			EnvVariables: map[string]string{
+				// GOOGLE_PROJECT unset
+				// GOOGLE_CLOUD_PROJECT unset
+				"GCLOUD_PROJECT":        "project-from-GCLOUD_PROJECT",
+				"CLOUDSDK_CORE_PROJECT": "project-from-CLOUDSDK_CORE_PROJECT",
+			},
+			ExpectedDataModelValue:    types.StringValue("project-from-GCLOUD_PROJECT"),
+			ExpectedConfigStructValue: types.StringValue("project-from-GCLOUD_PROJECT"),
+		},
+		"project value can be set by environment variable: CLOUDSDK_CORE_PROJECT is used fourth": {
+			ConfigValues: fwmodels.ProviderModel{
+				Project: types.StringNull(), // unset
+			},
+			EnvVariables: map[string]string{
+				// GOOGLE_PROJECT unset
+				// GOOGLE_CLOUD_PROJECT unset
+				// GCLOUD_PROJECT unset
+				"CLOUDSDK_CORE_PROJECT": "project-from-CLOUDSDK_CORE_PROJECT",
+			},
+			ExpectedDataModelValue:    types.StringValue("project-from-CLOUDSDK_CORE_PROJECT"),
+			ExpectedConfigStructValue: types.StringValue("project-from-CLOUDSDK_CORE_PROJECT"),
+		},
+		"when no project values are provided via config or environment variables, the field remains unset without error": {
+			ConfigValues: fwmodels.ProviderModel{
+				Project: types.StringNull(), // unset
+			},
+			ExpectedDataModelValue:    types.StringNull(),
+			ExpectedConfigStructValue: types.StringNull(),
+		},
+		// Handling empty strings in config
+		// TODO(SarahFrench) make these tests pass to address: https://github.com/hashicorp/terraform-provider-google/issues/14255
+		// "when project is set as an empty string the field is treated as if it's unset, without error": {
+		// 	ConfigValues: fwmodels.ProviderModel{
+		// 		Project: types.StringValue(""),
+		// 	},
+		// 	ExpectedDataModelValue:    types.StringNull(),
+		// 	ExpectedConfigStructValue: types.StringNull(),
+		// },
+		// "when project is set as an empty string an environment variable will be used": {
+		// 	ConfigValues: fwmodels.ProviderModel{
+		// 		Project: types.StringValue(""),
+		// 	},
+		// 	EnvVariables: map[string]string{
+		// 		"GOOGLE_PROJECT": "project-from-GOOGLE_PROJECT",
+		// 	},
+		// 	ExpectedDataModelValue:    types.StringNull(),
+		// 	ExpectedConfigStructValue: types.StringValue("project-from-GOOGLE_PROJECT"),
+		// },
+		// Handling unknown values
+		// TODO(SarahFrench) make these tests pass to address: https://github.com/hashicorp/terraform-provider-google/issues/14444
+		// "when project is an unknown value, the provider treats it as if it's unset (align to SDK behaviour)": {
+		// 	ConfigValues: fwmodels.ProviderModel{
+		// 		Project: types.StringUnknown(),
+		// 	},
+		// 	ExpectedDataModelValue:    types.StringNull(),
+		// 	ExpectedConfigStructValue: types.StringNull(),
+		// },
```

Comment:
I'll make these tests pass in a subsequent PR, for ease of review and to ensure that all these 'fixes' can be done at once later in a single PR instead of across multiple PRs